pool.go

```diff
@@ -50,6 +50,10 @@ type pool struct {
 	refresh chan struct{} // refresh is used to signal the need for doing a refresh of the Saturn endpoints pool.
 	done    chan struct{} // done is used to signal that we're shutting down the Saturn endpoints pool and don't need to refresh it anymore.
 
+	cidLk            sync.RWMutex
+	cidFailureCache  *cache.Cache // guarded by cidLk
```

Comment:
go-cache says:
> Its major advantage is that, being essentially a thread-safe map[string]interface{}

can we use it's internal thread safety rather than needing an explicit lock for accesses?

Reply:
@willscott I want both `cidFailureCache` and `cidCoolDownCache` to be updated atomically.